crates/fuzzing/src/generators/table_ops.rs

```diff
@@ -85,29 +117,22 @@ pub(crate) enum TableOp {
 }
 
 impl TableOp {
-    fn to_wat_string(&self, wat: &mut String) {
+    fn insert(&self, func: &mut Function) {
         match self {
             Self::Gc => {
-                wat.push_str("(call 0)");
+                func.instruction(Instruction::Call(0));
             }
             Self::Get(x) => {
-                wat.push_str("(drop (table.get $table (i32.const ");
-                write!(wat, "{}", x).unwrap();
-                wat.push_str(")))");
+                func.instruction(Instruction::Drop);
+                func.instruction(Instruction::TableGet { table: *x });
```

Comment:
These two instructions' order should be swapped. The result of the `table.get` is the thing we want to drop, so we want to do the `table.get` which pushes its result on top of the Wasm stack, and then do a `drop`, which pops the value off and drops it.

Reply:
OH that makes total sense -- I wasn't thinking in terms of putting things onto a stack first before using them